src/Sentry/Laravel/Tracing/Storage/TracingFilesystem.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace Sentry\Laravel\Tracing\Storage;
+
+use Illuminate\Contracts\Filesystem\Cloud as CloudFilesystemContract;
+use Illuminate\Contracts\Filesystem\Filesystem;
+use Illuminate\Filesystem\FilesystemAdapter;
+use Sentry\Tracing\SpanContext;
+use function Sentry\trace;
+
+class TracingFilesystem implements CloudFilesystemContract
+{
+    protected $filesystem;
+
+    public function __construct(Filesystem $filesystem)
+    {
+        $this->filesystem = $filesystem;
+    }
+
+    /**
+     * @template TResult
+     * @param array<string, mixed> $data
+     * @param callable(): TResult $originalCall
+     * @return TResult
+     */
+    protected function withTracing(string $method, array $data, callable $originalCall)
+    {
+        $context = new SpanContext();
+        $context->setOp("storage.{$method}");
+        $context->setData($data);
+
+        return trace($originalCall, $context);
+    }
+
+    public function assertExists($path, $content = null)
+    {
+        return $this->withTracing('assertExists', ['path' => $path], function () use ($path, $content) {
+            // TODO assertExists only present on FilesystemAdapter
+            return $this->filesystem->assertExists($path, $content);
+        });
+    }
+
```

Comment:
TODO add additional methods from `FilesystemAdapter` that are not part of `Illuminate\Contracts\Filesystem\Cloud`.